### PR TITLE
Updated Owners and Dependents page with glossary tooltip and reference links

### LINKED ISSUE #38136: Add references and glossary tooltip in the Owners and Dependents page
**This is a Feature Request**
**What would you like to be added**

- There are terms that can use the addition of a glossary tooltip to show the meaning of the said term.
- Some terms can add into What's next section like: ReplicaSets, DaemonSets, Deployments, Jobs and CronJobs, and ReplicationControllers.

**Why is this needed**
 This will enhance user experience and bring more clarity and less confusion.
**Comments**
Page to update: https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/

### PR DESCRIPTION
**Pull request by T-Lakshmi**
This PR updated `Owners and Dependents` page by adding few glossary tooltips and reference links to the terms which were not having previously. Also updated what's next section.

Fixes #38136.

### CODE REVIEW

**Comment by T-Lakshmi** [comment]
Hi @natalisucks, @shannonxtreme,
         PTAL!

**Comment by T-Lakshmi** [comment]
Hi @natalisucks, @shannonxtreme,
Do i need to add something else, WDYT?

File: content/en/docs/concepts/overview/working-with-objects/owners-dependents.md [hunk 1]
```diff
@@ -84,6 +84,8 @@ object.
...
 
-* Learn more about [Kubernetes finalizers](/docs/concepts/overview/working-with-objects/finalizers/).
-* Learn about [garbage collection](/docs/concepts/architecture/garbage-collection).
-* Read the API reference for [object metadata](/docs/reference/kubernetes-api/common-definitions/object-meta/#System).
\ No newline at end of file
+Learn more about the following:
+* [Kubernetes finalizers](/docs/concepts/overview/working-with-objects/finalizers/).
+* [garbage collection](/docs/concepts/architecture/garbage-collection).
+* API reference for [object metadata](/docs/reference/kubernetes-api/common-definitions/object-meta/#System).
+* Objets such as [ReplicaSets](/docs/concepts/workloads/controllers/replicaset/), [DaemonSets](/docs/concepts/workloads/controllers/daemonset/), [Deployments](/docs/concepts/workloads/controllers/deployment/), [Jobs](/docs/concepts/workloads/controllers/job/), [CronJobs](/docs/concepts/workloads/controllers/cron-jobs/) and [ReplicationControllers](/docs/concepts/workloads/controllers/replicationcontroller/).
```
Comment (shannonxtreme) [inline content/en/docs/concepts/overview/working-with-objects/owners-dependents.md:91]: Revert this change please, the original syntax is more readable

**Comment by shannonxtreme** [comment]
I'm fine with the glossary additions, although I wonder if we expect the reader to already know what things like Pods and namespaces are

File: content/en/docs/concepts/overview/working-with-objects/owners-dependents.md (hunk 1 above)
Comment (T-Lakshmi) [reply content/en/docs/concepts/overview/working-with-objects/owners-dependents.md:91]: Hi @shannonxtreme,
Here i added one extra point "line number 91" which were mention in page but no reference provided. 
Do you think it is not required to mention? Also advantage of this new syntax is avoiding the phrasing Learn/Read more about XYZ especially with the new addition.
WDYT? Please share your opinion.
 
**Preview Page**: https://deploy-preview-38227--kubernetes-io-main-staging.netlify.app/docs/concepts/overview/working-with-objects/owners-dependents/#what-s-next

Comment (shannonxtreme) [reply content/en/docs/concepts/overview/working-with-objects/owners-dependents.md:91]: I'd say revert it, I don't see a value add in this specific change @T-Lakshmi (including the new line). I understand your justification for the updated syntax, but IMO the What's Next section isn't like other lists - each line is independent of the others and are purely interest-based suggestions. the existing format gives us a lot more flexibility with wording as well.

Comment (T-Lakshmi) [reply content/en/docs/concepts/overview/working-with-objects/owners-dependents.md]: Hi @shannonxtreme,
Thanks for the suggestion. I reverted the changes, could you please take an another look!

**Comment by T-Lakshmi** [comment]
Hi @shannonxtreme,
Could you please take a look!

**Comment by shannonxtreme** [comment]
The referenced issue has not been triaged so I can't LGTM or approve this PR (ideally, you'd wait for the issue to be accepted before creating a PR). Let's wait for the issue to get triaged

**Comment by bradtopol** [comment]
@shannonxtreme @T-Lakshmi The issue has now been triaged.

**Comment by T-Lakshmi** [comment]
Hi @bradtopol,
Thanks for the confirmation. 

@shannonxtreme,

> The referenced issue has not been triaged so I can't LGTM or approve this PR (ideally, you'd wait for the issue to be accepted before creating a PR). Let's wait for the issue to get triaged

Thanks for the feedback, I will keep in mind next time.
As issue triaged now, could you please take another look into it.

**Comment by T-Lakshmi** [comment]
Hi,
Can anyone please take a look on it!
